BRANCHING_RELEASE_STRATEGY.md

```diff
@@ -0,0 +1,181 @@
+
+Start by being up-to-date to your `main` branch:
+```shell
+$ cd powsybl-repo
+$ git checkout main
+$ git pull
+```
+
+Create your temporary branch preparing to the release X.Y.0 and add commits bumping to your release version then your next snapshot version.
+You can then push this branch.
+```shell
+$ git checkout -b tmp_prepare_release
+$ mvn versions:set -DnewVersion=X.Y.0
+$ git commit -s -m "Bump to vX.Y.0"
+$ mvn versions:set -DnewVersion=X.Y+1.0-SNAPSHOT
+$ git commit -s -m "Bump to vX.Y+1.0-SNAPSHOT"
+$ git push -u origin tmp_prepare_release
+```
+
+Create a pull request from your temporary branch into the `main` branch and tag another maintainer as a reviewer so they can approve it.
+Once it is approved, merge it with the option `Rebase and merge`. **DO NOT squash the commit!**
+
+You can then create a Release note and tag via the GitHub UI, using the hash of the corresponding commit (bumping to vX.Y.0). The tag must respect the pattern `vX.Y.0`
+Please make sure that your release note is comprehensive to all new features and bug fixes of the release and that the migration guide has been updated if necessary.
+
+### Publishing a release
+
+**Before creating your local release, please make sure that your local repository state is clean.**
+
+On your repository, checkout to the release tag. You can then package and deploy your release:
+```shell
+$ git checkout tags/vX.Y.0
+$ mvn dependency:purge-local-repository
+$ mvn clean package -Prelease
+$ mvn deploy -Prelease -DskipTests
+```
+
+Your release will then be deployed in Sonatype. The complete documentation to release from your staging repository in Sonatype is detailed [here](https://central.sonatype.org/publish/release/#locate-and-examine-your-staging-repository).
+
+Once all the steps are completed, your release is published in maven central and might need a few more minutes to be available.
+
+If an issue occurs at any time during the releasing process, do not hesitate to check [Maven status](https://status.maven.org/) or to leave an issue on [Sonatype JIRA Dashboard](https://issues.sonatype.org/secure/Dashboard.jspa).
+
+### Differences for a corrective release
+
+Please note that there are some differences in the process when you're publishing a corrective release or a patch, which version respects the pattern `vX.Y.Z` with Z different from 0.
+
+First checkout to the previous `vX.Y.*` release or if a patch has already been released, on the `release-vX.Y.0` branch instead of the `main` branch.
+```shell
+$ git checkout tags/vX.Y.0
+$ git checkout -b release-vX.Y.0
+```
+or (if a patch has already been released)
+```shell
+$ git checkout release-vX.Y.0
+```
+
+You can then cherry-pick the commits of your patch:
+```shell
+$ git cherry-pick <commit1_hash>
+$ git cherry-pick <commit2_hash>
+...
+```
+And bump to the patched version:
+```shell
+$ mvn versions:set -DnewVersion=X.Y.Z
+$ git commit -s -m "Bump to X.Y.Z"
+$ git push -u origin release-vX.Y.0 --force-with-lease
```

Comment:
I cannot see why force would be needed, unless mistakes were made and need to be corrected

Reply:
I am not quite sure either. I can't think of a case where the `--force-with-lease` option is needed, but apparently some were already encountered (maybe to fix mistakes, as you suggested).
I propose to remove this option for now. If a releaser do really need it one day, it would remain possible to amend this document to add an explanation why in some cases this option could be needed.

Reply:
```suggestion
$ git push -u origin release-vX.Y.0
```